sci-libs/vtk/vtk-9.0.1.ebuild

```diff
@@ -139,6 +139,28 @@ DOCS=( CONTRIBUTING.md README.md )
 
 CHECKREQS_DISK_BUILD="3G"
 
+MY_CUDA_ARCH=
+
+get_cuda_arch() {
+	if [[ -z ${CUDA_ARCH} ]]; then
+		MY_CUDA_ARCH=native
+		return
+	else
+		# we ignore fermi arch, because current nvidia-cuda-toolkit-11*
+		# no longer supports it
+		case ${CUDA_ARCH} in
+			kepler|maxwell|pascal|volta|turing|ampere|all|none)
```

Comment:
Any chance to use `all` as default?

Automatic hardware detection is something that really should be avoided, no guarantee even being built on the system this will run on and so, unless user manually specify it through CUDA_ARCH, `all` seem like the only valid default.

I get that it'll make compilation take longer though (not that I tried it yet to see how bad it is).

Reply:
Build times will seriously hurt with `all`. While developing the ebuild, I had a build time of much more than a day

Reply:
Oh well. Alright then, keep native until a better solution -- at least have a einfo now so people are warned of this.

Not that einfo is very visible (won't show if people aren't looking at build.log), I'd say this is getting more important and maybe should be upgraded to ewarn if VTK_CUDA_ARCH is unset.

Reply:
fair enough

Reply:
What do you think of not respecting the `fermi` arch? We still have nvidia-cuda-toolkit v9 and v10 in the tree, which both support this arch? Is it better to support it, due to being outdated and probably not used with real cuda applications?

Reply:
Fermi needs LTS nvidia-drivers-390.xx and all in-tree cuda-toolkit currently have a dependency on DRIVER_PV >= 396.xx, meaning they can't install any of them as things stands, as such no sense in vtk worrying about fermi unless that's changed.

390 is also on its last legs, losing features after features (nvenc, uvm on x86, egl-wayland is broken too), and official end of support is due end of next year. I'll keep it in the tree for longer with minimal support but once something goes really wrong it'll be the end of it.

Reply:
> Automatic hardware detection is something that really should be avoided, no guarantee even being built on the system this will run on and so, unless user manually specify it through CUDA_ARCH, `all` seem like the only valid default.

What's wrong with introspection? We're not in the 1990's anymore. This should work in most of the cases.

Reply:
Notice how we also have CPU_FLAGS_X86, VIDEO_CARDS, etc... rather than auto-detection. The entire ecosystem is built around the user specifying what they can or can't use. This is a bit like a build system passing -march=native by default and later failing on everything but the current hardware.

Reply:
Your examples are different. We don't have a USE_EXPAND for CUDA architectures yet. I agree, that it's better to choose from a pre-defined set of values, yet we don't have such values. There's nothing wrong with using -march=native, I just have to be aware of the consequences and can not expect a binary built with it, to be able to run on another hardware. But I fully trust the compiler to produce valid code which can run on my machine. The same is true for the GPU built into my machine.

Reply:
I thought we had already concluded that was okay going with native due to build times? I'm not sure why you brought this thread back up.

We could keep arguing about how this goes against Gentoo philosophy but this doesn't change the ebuild unless you're trying to get convinced to go back on this.

Reply:
> I'm not sure why you brought this thread back up.

Gonna write you in private about this.